testgrid/cmd/transfigure/cmd/transfigure.go

```diff
@@ -0,0 +1,363 @@
+	rootCmd.PersistentFlags().StringVar(&remoteForkRepo, "remote_fork_repo", "", "The name of the user's fork of kubernetes/test-infra, if it is not named test-infra.")
+	rootCmd.PersistentFlags().StringVar(&gitUser, "git_user", "", "The GitHub username to use. Pulled from github_token if not specified.")
+	rootCmd.PersistentFlags().StringVar(&gitEmail, "git_email", "", "The GitHub email to use. Pulled from github_token if not specified.")
+	rootCmd.PersistentFlags().BoolVar(&dryRun, "dry_run", false, "If true, no PR will be created.")
+}
+
+func createTempWorkingDir() error {
+	tempDir, err := ioutil.TempDir(".", "transfigure_")
+	if err != nil {
+		return wrapErrorOrNil("Error creating temp dir", err)
+	}
+	workingDir, err = filepath.Abs(tempDir)
+	if err != nil {
+		return wrapErrorOrNil("Error getting abs path for working dir", err)
+	}
+	log.Printf("Created temp directory %v/", tempDir)
+	return nil
+}
+
+func removeTempWorkingDir() {
+	os.RemoveAll(workingDir)
+	log.Printf("Removed temp directory %v/", workingDir)
+}
+
+func cloneK8sTestInfraRepo() error {
+	log.Printf("Cloning kubernetes/test-infra into temp directory")
+	cmd := exec.Command("git", "clone", "https://github.com/kubernetes/test-infra.git")
+	_, err := runCmd(cmd)
+	return wrapErrorOrNil("Error on git clone", err)
+}
+
+func createAndCheckoutGitBranch() error {
+	log.Print("Checking out branch " + Branch)
+	_, err := runCmd(exec.Command("git", "checkout", "-B", Branch))
+	return wrapErrorOrNil("Error on git checkout -B "+Branch, err)
+}
+
+func createRepoSubdirForTestgridYAML() {
+	testgridDir = filepath.Join(workingDir, "test-infra/config/testgrids", repoSubdir)
+	if _, err := os.Stat(testgridDir); os.IsNotExist(err) {
+		log.Printf("Directory %v does not exist; creating it.", testgridDir)
+		os.MkdirAll(testgridDir, 0755)
+	}
+}
+
+func generateTestgridYAML() error {
+	yamlPath, err := filepath.Abs(testgridYaml)
+	if err != nil {
+		return wrapErrorOrNil("Invalid testgrid yaml path "+testgridYaml, err)
+	}
+
+	cmd := exec.Command(
+		"/Users/joshuabone/go/src/k8s.io/test-infra/testgrid/cmd/transfigure/configurator",
+		fmt.Sprintf("--prow-config=%s", prowConfig),
+		fmt.Sprintf("--prow-job-config=%s", prowJobConfig),
+		"--output-yaml",
+		fmt.Sprintf("--yaml=%s", yamlPath),
+		"--oneshot",
+		fmt.Sprintf("--output=%s/gen-config.yaml", testgridDir),
+	)
+	_, err = runCmd(cmd)
+	return wrapErrorOrNil("Error running configurator", err)
+}
+
+func gitAddAll() error {
+	_, err := runCmd(exec.Command("git", "add", "--all"))
+	return wrapErrorOrNil("Error on git add", err)
+}
+
+func gitDiffExists() bool {
+	cmd := exec.Command("git", "diff", "--cached", "--quiet", "--exit-code")
+	cmd.Dir = workingDir
+	err := cmd.Run()
+	return err != nil
+}
+
+func runBazelTests() error {
+	log.Print("Running kubernetes/test-infra tests...")
+	cmd := exec.Command("bazel", "test", "test-infra/config/tests/...", "test-infra/hack:verify-spelling")
+	out, err := runCmd(cmd)
+	if err != nil {
```

Comment:
Why not simply return err? `out` should be printed no matter there is error or not imo

Reply:
Done.